mithril-test-lab/mithril-end-to-end/src/end_to_end_spec.rs

```diff
@@ -1,22 +1,29 @@
-use crate::assertions;
-use crate::MithrilInfrastructure;
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use tokio::sync::RwLock;
+use tokio::task::JoinSet;
+
 use mithril_common::{
+    chain_observer::ChainObserver,
     entities::{Epoch, SignedEntityTypeDiscriminants},
     StdResult,
 };
 
-pub struct Spec<'a> {
-    pub infrastructure: &'a mut MithrilInfrastructure,
+use crate::{assertions, Aggregator, MithrilInfrastructure};
+
+pub struct Spec {
+    pub infrastructure: Arc<RwLock<Option<MithrilInfrastructure>>>,
```

Comment:
The fact that the `main.rs` use a `Arc<RwLock<Option<_>>>` to handle the infrastructure lifecycle should not appears here, instead we should have a `Arc<MithrilInfrastructure>`.

Reply:
You're right, thanks for creating the fix :+1: